checker.c

```diff
@@ -1,21 +1,164 @@
+ * if the current SOH rating us below the threshold 0.5%, then the battery is unacceptable.
+ * The SOH at the manufacturing unit is 100%
+ * input: Current SOH value
+ * returns: Check if the value is greater than the threshold
+ *********************************************************************************/
+int StateOfHealth_check (float currentsoh)
+{
+  return (currentsoh > 0.5 ? 1:0);
+}
+
+/*************************************************************************************************
+ * Process: Display the State of Health of the battery currently as compared to ideal conditions
+ ************************************************************************************************/
+int BMS_StateOfHealth (float soh)
+{
+ 
+ int sohcheck = StateOfHealth_check(soh);
+ if (sohcheck)
+  {
+    printf(" State-of-Health of battery is %f, Battery conditions are good as compared to ideal conditions \n", soh );
+	return 1;
+  }
+  else
+  {
+    printf (" State-of-Health of battery is %f, Battery conditions are poor and cannot be used for the application \n", soh);
+	return 0;
+  } 
+}
+
+
+/********************************************************************************
+ * A function that gives Charge rate of a Battery management system.
+ * if the current Charge rating is above the threshold, then the battery is unacceptable.
+ * input: Current charge rate in decimal (percentage converted to floating value)
+ * returns: Check if the charge rate is out of boundary conditions
+ *********************************************************************************/
+int BMS_ChargeRateCheck(float charge_rate)
+{
+	int chargerate_check=(charge_rate>MAXCHARGERATE);
+	 if(chargerate_check)
+	   {
+		printf("Charge Rate is %f and is out of range!\n", charge_rate);
+		return 0;
+	   }
+	printf("Charge Rate is %f within the maximum threshold\n", charge_rate);  
+	return 1;
+}
+/********************************************************************************
+ * A common function that checks the range of parameters.
+ * input: parameter, aximum and minimum range to be checked
+ * returns: Check if the parameter is out of the given maximum and minimum range
+ *********************************************************************************/
+bool BMS_RangeCheck(float parameter, float maxrange, float minrange)
+{
+	return((parameter < minrange) || (parameter > maxrange));
+}
+/********************************************************************************
+ * A function that gives State-of-Charge parameter check of a Battery management system.
+ * if the current SOC is outside the boundary conditions, then the battery is unacceptable.
+ * if the SOC exceeds the 80% threshold, that reduces the life span of the battery, and losses are incurred
+ * Battery is charged above 80% only in outstation charging system.
+ * input: SOC in percentage
+ * returns: Check if the SOC is out of boundary conditions
```

Comment:
Good comment to express that `BMS_StateOfCharge` returns 0 when things are ok, and 1 when something is wrong.
Can you carry this meaning in the function name, so that people reading the code that calls this function need not search the comment?